pyflakes/test/test_type_annotations.py

```diff
@@ -300,6 +308,13 @@ class A:
         class B: pass
         ''', m.UndefinedName)
 
+        self.flakes('''
+        from __future__ import annotations
+        T: object
+        def f(t: T): pass
+        def g(t: 'T'): pass
+        ''')
+
```

Comment:
one test I'd be interested in is:

```python
def f():
    x_is_unused: int
```

Reply:
I have added this test now. This is actually something I noticed while implementing this. Would it make sense to change the current behavior (no error) in a future PR to flag this? Outside modules and class definitions this looks like an oversight.

Reply:
hmmm interesting, yeah I was hoping it would flag it similar to the unused local

```console
$ ~/opt/venv/bin/pyflakes
def f():
    x_is_unused = 1
<stdin>:2: local variable 'x_is_unused' is assigned to but never used
```

but afaict it's not a regression and can be done later

___

I've definitely seen cases where annotate-then-assign is helpful, especially with really long types so it would only really make sense if it's annotated and then never used

```python
self.thing: Dict[str, Set[Tuple[int, str]]]
self.thing = collections.defaultdict(set)
```

Reply:
so maybe a TODO comment that this should be improved in the future 👍

Reply:
I have added a few more tests and a TODO comment.